src/neptune_scale/__init__.py

```diff
@@ -611,6 +627,11 @@ def _wait(
                     wait_condition.wait(timeout=wait_time)
                     value = wait_value.value
 
+                timeout_left -= time.monotonic() - t0
```

Comment:
why are you doing it this way instead of just 

```
start_time = time.monotonic()
while True:
    ..
    time_elapsed = time.monotonic() - start_time
    if time_elapsed > timeout:
        boom
```

Your implementation may wait longer than the specified timeout timeout, because you are moving the t0 on each loop and thus ignoring the time of executing BLOCK B 

```
while True:
    t0 = time.monotonic()   ## stores e.g.   1000, 2000, 3000   on each iteration
    BLOCK A
    timeout_left = time.monotonic() - t0   ## subtracts e.g. 1900 - 1000, 2900 - 2000, 3900 - 200
    BLOCK B
```